### PR TITLE
3 create hero section of landing page

### LINKED ISSUE #3: Create Hero section of Home page
### Figma link for component:
[Hero section on Figma](https://www.figma.com/file/QWycOkxNJmtiyZ3v8imIzx/Code-PDX-website?type=design&node-id=133-397&mode=design&t=JhDPDVeP1SwPMWwa-4)

---
### Feature Description:
Create Hero section of homepage containing:
- Header text
- Contact Us button
- St. John's Bridge photo

---
### Screenshots:
Screenshot 2023-10-04 at 8 19 28 PM

### PR DESCRIPTION
**Pull request by ogorman89**
## This PR:

Resolves #3 

Adds a basic version of the responsive hero section to Home page. The breakpoints are abrupt so they will need to be adjusted.

---

## Screenshots (if applicable):

https://github.com/codeforpdx/codepdx_website/assets/3335277/fa00e362-474c-4450-bd9b-8ce27e7a95a7

---

## Future Steps/PRs Needed to Finish This Work (optional):

We will need to adjust breakpoints for the desired final state mobile layout.

### CODE REVIEW

**Review by andycwilliams — CHANGES_REQUESTED** [review]
Good start! Just a couple of things. 

---

Be sure to selecting the development branch to merge into rather than main.

---

On some screen sizes, the contact us button doesn't scale properly. This can be addressed by adjusting the width in `contactBtnStyle` on line 37.

Capture
Capture2
Capture3

---

Also, because the branch name only mentions the landing page, does this mean the hero sections for the other pages will be in a separate PR? 

If so, that's fine. It only means issue 3 wouldn't be closed by this PR alone.

**Comment by ogorman89** [comment]
@andycwilliams I fixed the PR to reference the `development` branch. Good catch! 

I can't reproduce the button issues you screenshot. What browser were you testing in?

From @xscottxbrownx Issue 3:
> Create Hero section of homepage containing:
> 
> Header text
> Contact Us button
> St. John's Bridge photo

I was planning on opening a new issue for the next step which is either:
1. A reusable hero section, that can be formatted and sized
2. Stand-alone hero sections for the other two pages

**Comment by andycwilliams** [comment]
I used Brave, which is Chromium-based, but also tried Chrome and Edge and got the same results. On some of the preset sizes it was fine, but on others the text got skewed like in the screenshots. It shouldn't take much to fix, though (knock on wood).

Oops. I just assumed Issue 3 covered all the hero sections. In that case this PR is enough but we need another issue written up.

e: Also tried Firefox and DuckDuckGo. Same deal.

**Comment by Jared-Krajewski** [comment]
> was planning on opening a new issue for the next step which is either:
> 
> 1. A reusable hero section, that can be formatted and sized
> 2. Stand-alone hero sections for the other two pages

that approach sounds good to me.

**Comment by xscottxbrownx** [comment]
My thoughts, 

This is a great start for first time coding in React!

---

1. we haven't had too many PR's on this project yet - so haven't talked much about code consistency/styling.

**This is a personal opinion, and ultimately doesn't matter for code to work, but:**
I'd prefer if we are going to use styling like this (declared separately as an object/styled components/etc), that we move that to the bottom of the file. I'd prefer to see what this file/component is and does, before seeing the styling for each piece of it.
So, this for example:
```js
// Material UI Imports
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

const Hero = () => {
  return (
    <Box sx={heroBackground}>
      <Box sx={heroContainer}>
        <Typography variant="h1" component="h1" sx={h1Style}>
          Bridging Technology and Civil Services
        </Typography>
        <Button
          variant="contained"
          href="mailto:hugh@codeforpdx.org"
          target="_blank"
          rel="noopener"
          sx={contactBtnStyle}
        >
          Contact Us
        </Button>
      </Box>
    </Box>
  );
};

export default Hero;

const heroBackground = {
  width: '100vw',
  height: { xs: '30vh', md: '60vh' },
  backgroundImage: 'url(/assets/stJohnsBridge.png)',
  backgroundSize: 'cover',
  backgroundPosition: 'center',
  backgroundRepeat: 'no-repeat'
};

const heroContainer = {
  display: 'flex',
  flexFlow: 'column',
  flexBasis: 1,
  justifyContent: 'flex-start',
  alignItems: { xs: 'center', md: 'flex-start' },
  width: { xs: '100%', md: 3 / 5 }
};

const h1Style = {
  fontSize: { xs: '6vw', md: '6vw' },
  pt: '1em',
  pl: '0.75em',
  width: { xs: '80%', md: '100%' },
  textAlign: 'center'
};

const contactBtnStyle = {
  alignSelf: 'center',
  mt: '2em',
  height: '2em',
  width: { xs: '18vh', md: '17vh' }
};
```

---

2. For MUI `Typography` if the `variant` and `component` are the same, you don't need the `component`.

The `variant` makes that component, only needed when you want to style it like a specific variant, but want it to be a different component. If you take away the `component="h1"` and inspect page, you'll see it's still an `h1` in the DOM.

---

3. Couple of styling issues that are noticeable to end-user.

- certain screen size, the button floats out of the `Hero` section:
Screenshot 2023-10-21 at 7 49 05 AM

- at smaller screen sizes, the button becomes perceived as more important due to being more prominent than the `h1`
Screenshot 2023-10-21 at 7 49 57 AM

- readability of `h1`
   Think it needs a text-shadow to more closely match Figma file, and be easier to read. 
    If unfamiliar with that, see something basic [here](https://www.w3schools.com/cssref/css3_pr_text-shadow.php) and click `show demo`

**Comment by ogorman89** [comment]
@andycwilliams
1. Button should be fixed
2. As for the button overflowing the container. I made some changes, so this should no longer happen for common viewports. Note: You can still find odd-shaped window sizing that reproduces this behavior. 

I think this can be completely resolved if I go fully responsive. I have many px-sized elements in the current solution. If we want full responsivity, I probably need to refactor with relative units for everything. I'd probably defer doing that until we have the design for the mobile layout.

@xscottxbrownx 
1. As requested, I have moved my style definitions to the end of the file.
2. I removed the component reference and kept the variant. 
3. I have mostly resolved this issue, you can still create the overflowing effect for odd sized windows. See 2. in my response to @andycwilliams above.
4. I've re-styled the h1 to more closely match the figma board. 
- The desktop view should be very similar now. 
- I've added the drop shadow that is in the figma mockup for both the h1 and button
- The button has been scaled down in mobile (not quite sure what we want mobile to look like yet)

TLDR;
It's better, but there are still some issues. It should look ok on common viewports. 

Pictures
Desktop
Screen Shot 2023-10-22 at 1 25 46 PM

SE
Screen Shot 2023-10-22 at 1 27 57 PM

iPad Mini (this looks bad)
Screen Shot 2023-10-22 at 1 30 11 PM

Weird size to reproduce the overflow 
Screen Shot 2023-10-22 at 1 31 39 PM

**Comment by andycwilliams** [comment]
Looking better! Button text is fixed and this PR should be close to ready.

You've mentioned the next steps to take but I wanted to show how it looks on my end.

If you're going to wait to finetune styling in a future PR, then we should get a new issue up with the details. Then we can close this one out.

Capture

The only points I'll mention at this time is that on smaller screens the `h1` text becomes hard to read. We could maybe just make a conditional text shadow for mobile screens, one that's lighter instead of darker.

Also, I'd prefer a bit less padding on the desktop version of the button. But that's more subjective and I'd wait for more work on the Figma board first.

And if as for moving styling to the bottom, we ought to figure out for sure what we want to do. We should be consistent and apply whatever we decide to the whole app.

Honestly, I'm not entirely sure what viewports we can just safely ignore, if any. I think it'll depend on feedback and what case workers actually (and most commonly) use.

**Comment by xscottxbrownx** [comment]
Getting there.  

We don't have mobile designs yet, but for button going over container issue - what do we think about just making it's height `100vh`? That would be my go-to and solution for a splash/intro/home page...

Anybody feel like chiming in on that ⬆️ ?   I don't know what was envisioned for size of that section honestly.

---

I'm with Andy on having less padding on button on desktop. Seems disproportionately large to me. Like that is the only call to action almost.  I'd prefer most people scroll down over clicking that button.   We *could* do a bouncing down arrow icon in the odot green color, small towards bottom center if we want.    Or figure out the image/button/container sizing issue and leave the "Volunteer with Us" above-the-fold, so users can easily tell there's more content to scroll to.

**Comment by Jared-Krajewski** [comment]
the hero image in the Figma has a contrast change to make the black text more readable. Not perfect but might help with that visibility.  that weird size will likely never be seen and we can re evaluate these funky sizes in the future if they become a problem. Im OK with the button being a bit smaller. 

 Im OK with this as is but would be nice to see the background image faded a bit to help the contrast in those other sizes where text goes to the dark areas of the image, again can be changed after another Figma session.

**Comment by ogorman89** [comment]
> Looking better! Button text is fixed and this PR should be close to ready.
> 
> You've mentioned the next steps to take but I wanted to show how it looks on my end.
> 
> If you're going to wait to finetune styling in a future PR, then we should get a new issue up with the details. Then we can close this one out.
> 
> Capture
> The only points I'll mention at this time is that on smaller screens the `h1` text becomes hard to read. We could maybe just make a conditional text shadow for mobile screens, one that's lighter instead of darker.
> 
> Also, I'd prefer a bit less padding on the desktop version of the button. But that's more subjective and I'd wait for more work on the Figma board first.
> 
> And if as for moving styling to the bottom, we ought to figure out for sure what we want to do. We should be consistent and apply whatever we decide to the whole app.
> 
> Honestly, I'm not entirely sure what viewports we can just safely ignore, if any. I think it'll depend on feedback and what case workers actually (and most commonly) use.

@andycwilliams I think Jared’s comment on the image contrast change should help with readability so I’ll start with that. 

As for button styling, let’s revise the figma mockup.

**Comment by ogorman89** [comment]
> the hero image in the Figma has a contrast change to make the black text more readable. Not perfect but might help with that visibility. that weird size will likely never be seen and we can re evaluate these funky sizes in the future if they become a problem. Im OK with the button being a bit smaller.
> 
> Im OK with this as is but would be nice to see the background image faded a bit to help the contrast in those other sizes where text goes to the dark areas of the image, again can be changed after another Figma session.

@Jared-Krajewski My mistake I missed the contrast adjustment on the image. I will fix that.

**Comment by ogorman89** [comment]
> Getting there.
> 
> We don't have mobile designs yet, but for button going over container issue - what do we think about just making it's height `100vh`? That would be my go-to and solution for a splash/intro/home page...
> 
> Anybody feel like chiming in on that ⬆️ ? I don't know what was envisioned for size of that section honestly.
> 
> I'm with Andy on having less padding on button on desktop. Seems disproportionately large to me. Like that is the only call to action almost. I'd prefer most people scroll down over clicking that button. We _could_ do a bouncing down arrow icon in the odot green color, small towards bottom center if we want. Or figure out the image/button/container sizing issue and leave the "Volunteer with Us" above-the-fold, so users can easily tell there's more content to scroll to.

@xscottxbrownx 100vh would be the easiest solution but on larger phone viewports it looked a little weird to me. Either way we should hash it out in figma and figure out how we want things to scale for large mobile viewports. 

Generally I’d agree on less padding on the button I was attempting to match the figma file. I think the button looks good with the extra padding on desktop but I don’t have a strong opinion either way. Let’s play with the figma file and then adjust the button.

**Comment by andycwilliams** [comment]
I'd like to see how this looks when merged with the most recent changes. I don't think then it'll take too much more to make it ready to approve.

**Comment by Jared-Krajewski** [comment]
@ogorman89 

line 30: 
```  
backgroundImage: `linear-gradient(to top, rgba(255, 255, 255, 0.1), rgba(255, 255, 255, 0.3)), url(/assets/stJohnsBridge.png)`,
```
Screenshot 2023-10-23 at 4 50 31 PM

I'm sure we might be able to increase readability in the future but after design session.

**Review by andycwilliams — APPROVED** [review]

**Review by Jared-Krajewski — APPROVED** [review]
